core/src/main/java/org/infinispan/factories/EmptyConstructorNamedCacheFactory.java

```diff
@@ -114,9 +115,12 @@ public Object construct(String componentName) {
       } else if (componentName.equals(CommandsFactory.class.getName())) {
          return new CommandsFactoryImpl();
       } else if (componentName.equals(PersistenceManager.class.getName())) {
-         PersistenceManagerImpl persistenceManager = new PersistenceManagerImpl();
+         PersistenceManager persistenceManager = new PersistenceManagerImpl();
+         if (globalConfiguration.tracing().enabled()) {
+            persistenceManager = new TracedPersistenceManager(persistenceManager);
```

Comment:
no idea why we decided to decorate instead of implementing the changes in PersistenceManagerImpl. The changes look pretty small to me.

Reply:
I personally think separating out the tracing code to not be in the actual impl code is a lot cleaner than living inside. However, now that we are trimming it to be much less on the happy path, I agree it can probably live in the impl. It will increase the loaded classes size a tiny bit when enabled, but that is probably fine in the grand scheme

Reply:
my other concern, if we keep this,  is that the TracedPersistenceManager is between Passivation and PersistenceManagerImpl

Reply:
@wburns / @fax4ever  offtopic, have you had the chance to check the performance impact with and without tracing?

Reply:
> my other concern, if we keep this, is that the TracedPersistenceManager is between Passivation and PersistenceManagerImpl

There shouldn't be any issue there. Passivation has to be outside of trace wrapper otherwise the call to passivate wouldn't be caught by the tracing.

Reply:
Thank you for checking it @wburns. I resolve this conversation.